src/Hl7.Fhir.Specification/Validation/OperationOutcomeExtensions.cs

```diff
@@ -32,6 +34,42 @@ public static void Flatten(this OperationOutcome outcome)
                 issue.RemoveExtension(ValidationOutcomeExtensions.OPERATIONOUTCOME_ISSUE_HIERARCHY);
             }
         }
+
+        public static OperationOutcome RemoveDuplicateMessages(this OperationOutcome outcome)
+        {
+            var comparer = new IssueComparer();
+            outcome.Issue = outcome.Issue.Distinct(comparer).ToList();
+            return outcome;
+        }
+    }
+
+    internal class IssueComparer : IEqualityComparer<OperationOutcome.IssueComponent>
+    {        
+        public bool Equals(OperationOutcome.IssueComponent x, OperationOutcome.IssueComponent y)
+        {
+            if (x is null && y is null)
+                return true;
+            else if (x is null || y is null)
+                return false;
+            else if (x.Location?.FirstOrDefault() == x.Location?.FirstOrDefault() && x.Details?.Text == y.Details?.Text)
+                return true;
+            else
+                return false;
+        }
+
+        public int GetHashCode(OperationOutcome.IssueComponent issue)
+        {
+            var hash = issue?.Location?.FirstOrDefault()?.GetHashCode() ^ issue?.Details?.Text?.GetHashCode();
```

Comment:
There has been a lot of discussion about getting the right Hashcode on the internet. This is one implementation. Using the bitwise xor operation (^) can cause that the value is overflowing (does not fit into an int anymore). For that matter you can use the unchecked statement to prevent that so:

```suggestion
            var hash = unchecked(issue?.Location?.FirstOrDefault()?.GetHashCode() ^ issue?.Details?.Text?.GetHashCode());
```